### PR TITLE
Fix an issue where Vercel adapter may create functions for prerendered routes

### LINKED ISSUE #10211: Vercel adapter: parameter "functionPerRoute" generate serverless function even for prerendered routes
### Astro Info

```block
Astro                    v4.4.2
Node                     v20.10.0
System                   macOS (arm64)
Package Manager          yarn
Output                   hybrid
Adapter                  @astrojs/vercel/serverless
Integrations             @astrojs/tailwind
                         astro-icon
                         @astrojs/vue
                         @storyblok/astro
                         extract-tailwind-classes
                         redirections
```

### If this issue only occurs in one browser, which browser is a problem?

All

### Describe the Bug

Actually, when I build the projet with the Vercel adapter and the parameter `functionPerRoute` to `true`, it generates serverless functions for all routes, prerendered ones too. I have a route named [...slug].astro that generates only prerendered pages. But with this configuration it generates a serverless function [...slug].astro, as the result, all 404 errors fallback on this serverless function and a response is returned with code 200. If i set `functionPerRoute`to `false`, it's works as intended.

### What's the expected result?

Must not generate serverless functions for prerendered pages.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-9beo4b

### Participation

- [ ] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by mingjunlu**
## Changes

- Filters out prerendered routes for Vercel adapter
- Closes #10211

## Testing

- Test cases updated

## Docs

- N/A

### CODE REVIEW

File: packages/integrations/vercel/src/serverless/adapter.ts [hunk 1]
```diff
@@ -288,7 +288,10 @@ export default function vercelServerless({
 				}
 			},
 			'astro:build:ssr': async ({ entryPoints, middlewareEntryPoint }) => {
-				_entryPoints = entryPoints;
+				const entryPointsWithoutPrerenderedRoutes = new Map(
+					Array.from(entryPoints).filter(([routeData]) => !routeData.prerender)
+				);
+				_entryPoints = entryPointsWithoutPrerenderedRoutes;
```
Comment (ematipico) [inline packages/integrations/vercel/src/serverless/adapter.ts:294]:
```suggestion
				_entryPoints = new Map(
					Array.from(entryPoints).filter(([routeData]) => !routeData.prerender)
				);
```

**Review by ematipico — APPROVED** [review]

File: packages/integrations/vercel/src/serverless/adapter.ts (hunk 1 above)
Comment (mingjunlu) [reply packages/integrations/vercel/src/serverless/adapter.ts]: Thanks for the quick review, @ematipico! I've applied the suggested change 😊

File: .changeset/four-shoes-rule.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@astrojs/vercel": patch
+---
+
+Fixes an issue where Vercel adapter may create functions for prerendered routes
```
Comment (lilnasy) [inline .changeset/four-shoes-rule.md:5]:
```suggestion
Fixes an issue where functions were also created for prerendered routes with `functionPerRoute` enabled.
```

**Review by lilnasy — APPROVED** [review]
Looks great to me! Thanks for the clear tests as well.

**Pull request merged**